backend/service/github/github.go

```diff
@@ -236,6 +242,25 @@ func (s *svc) CreatePullRequest(ctx context.Context, ref *RemoteRef, base, title
 	}, nil
 }
 
+func (s *svc) ListPullRequestsWithCommit(ctx context.Context, ref *RemoteRef, sha string) ([]*PullRequestInfo, error) {
```

Comment:
i would expand the signature a bit so this could be used to query for more than just open PRs in the future without a breaking signature change.

fake edit: according to https://docs.github.com/en/rest/reference/repos#list-pull-requests-associated-with-a-commit it doesn't accept `state` as a parameter though, so maybe the comment below is incorrect.

Reply:
Good point. Upon looking again, seems like Github has some contradicting information. 

Here (https://github.com/google/go-github/blob/ff33a554ef2c24c1710238b69942fe51e70d89da/github/pulls.go#L121-L123) PullRequestListOptions, which is the opts part of `ListPullRequestsWithCommit(ctx context.Context, owner, repo, sha string, opts *githubv3.PullRequestListOptions)` stats that Default is "open". 

However, the method comment here https://github.com/google/go-github/blob/master/github/pulls.go#L168-L194 seems to imply that the default is open and closed. 

I'll see if I can find out more.

Reply:
Following up with go-github here https://github.com/google/go-github/pull/2228

Reply:
I think Daniel was suggesting supporting either `PullRequestListOptions` in the signature so that callers could specify this.

Reply:
ahh I see, I jumped the gun too quickly 

done -- added in `state string` into the signature so callers the githubv3 `PullRequestListOptions` struct is abstracted away for callers. wdyt?

Reply:
imo the PR list options object is gives the most flexibility for users of the service without having to change / amend the function signature.

Reply:
Ah ok fair point. Done -- d36be40